### PR TITLE
Added validation

### PR DESCRIPTION
**Pull request by ManishParyani**
### JIRA link (if applicable) ###
RDCC-4137
[RDCC-4137](https://tools.hmcts.net/jira/browse/RDCC-4137)

### Change description ###
Added validation to improve performance

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

**Review by shivajihmcts — DISMISSED** [review]

File: src/functionalTest/java/uk/gov/hmcts/reform/judicialapi/JudicialUsersFunctionalTest.java [hunk 1]
```diff
@@ -101,33 +98,26 @@ void shouldGet403WhenUserSearchApiToggledOff() {
...
 
-        if (OK.value() == refreshResponse.getStatusCode()) {
-            List<UserProfileRefreshResponse> userProfiles = asList(refreshResponse.getBody()
-                    .as(UserProfileRefreshResponse[].class));
-            log.info("JRD get refreshResponse response: {}", userProfiles.get(0).getObjectId());
-            assertNotNull(userProfiles.get(0).getObjectId());
-        } else {
-            assertEquals(NOT_FOUND.value(),refreshResponse.getStatusCode());
-        }
+        assertEquals(BAD_REQUEST.value(),refreshResponse.getStatusCode());
```
Comment (SujeethKumarAri) [inline src/functionalTest/java/uk/gov/hmcts/reform/judicialapi/JudicialUsersFunctionalTest.java:117]: Looks like you have removed a success scenario and replaced with negative scenario. Now there are no success scenarios for refresh api. Can you add this as a separate test ?

File: src/functionalTest/java/uk/gov/hmcts/reform/judicialapi/JudicialUsersFunctionalTest.java [hunk 2]
```diff
@@ -143,15 +133,7 @@ void refreshUserProfileSortDesc(String role) {
...
-        if (OK.value() == refreshResponse.getStatusCode()) {
-            List<UserProfileRefreshResponse> userProfiles = asList(refreshResponse.getBody()
-                    .as(UserProfileRefreshResponse[].class));
-
-            log.info("JRD get refreshResponse response: {}", userProfiles.get(0).getObjectId());
-            assertNotNull(userProfiles.get(0).getObjectId());
-        } else {
-            assertEquals(NOT_FOUND.value(),refreshResponse.getStatusCode());
-        }
+        assertEquals(BAD_REQUEST.value(),refreshResponse.getStatusCode());
```
Comment (SujeethKumarAri) [inline src/functionalTest/java/uk/gov/hmcts/reform/judicialapi/JudicialUsersFunctionalTest.java:136]: Looks like you have removed a success scenario and replaced with negative scenario. Now there are no success scenarios for refresh api. Can you add this as a separate test ?

File: src/integrationTest/java/uk/gov/hmcts/reform/judicialapi/RefreshUserProfileIntegrationTest.java [hunk 3]
```diff
@@ -99,12 +100,11 @@ void shouldReturn_200_ValidParameters_ccdPageSize(String role) {
...
                 .build();
         // pageSize 2
-        var response = judicialReferenceDataClient.refreshUserProfile(refreshRoleRequest,2,
+        var errorResponseMap = judicialReferenceDataClient.refreshUserProfile(refreshRoleRequest,2,
                 0,"", "objectId", role, false);
-        assertThat(response).containsEntry("http_status", "200 OK");
-
-        var userProfileList = (List<?>) response.get("body");
-        assertThat(userProfileList).hasSize(2);
+        assertThat(errorResponseMap).containsEntry("http_status", "400");
```
Comment (SujeethKumarAri) [inline src/integrationTest/java/uk/gov/hmcts/reform/judicialapi/RefreshUserProfileIntegrationTest.java:105]: Looks like you have removed a success scenario and replaced with negative scenario. Now there are no success scenarios for refresh api. Can you add this as a separate test ?

File: src/functionalTest/java/uk/gov/hmcts/reform/judicialapi/JudicialUsersFunctionalTest.java (hunk 2 above)
Comment (ManishParyani) [reply src/functionalTest/java/uk/gov/hmcts/reform/judicialapi/JudicialUsersFunctionalTest.java:136]: We can add success scenario but that would create dependency on database, if anybody update data or remove data in the database then our test case will start failing.

Comment (SujeethKumarAri) [reply src/functionalTest/java/uk/gov/hmcts/reform/judicialapi/JudicialUsersFunctionalTest.java:136]: that's why we have added these
`if (OK.value() == refreshResponse.getStatusCode()) {
            List<UserProfileRefreshResponse> userProfiles = asList(refreshResponse.getBody()
                    .as(UserProfileRefreshResponse[].class));
            log.info("JRD get refreshResponse response: {}", userProfiles.get(0).getObjectId());
            assertNotNull(userProfiles.get(0).getObjectId());
        } else {
            assertEquals(NOT_FOUND.value(),refreshResponse.getStatusCode());
        }`

Comment (ManishParyani) [reply src/functionalTest/java/uk/gov/hmcts/reform/judicialapi/JudicialUsersFunctionalTest.java:140]: Yeah It make sense, I have added one success scenario as well. Thanks for the clarification.

**Review by SujeethKumarAri — APPROVED** [review]
Thanks @ManishParyani for adding the tests

**Review by shivajihmcts — APPROVED** [review]

**Pull request merged**